src/main/kotlin/traffic_simulation/Main.kt

```diff
@@ -65,16 +67,28 @@ fun parseInputOfCSV ( fileName : String ): MutableList<Vehicle>{
 
     // insert the parsed information of csv-file in usable lists and use them in functions
     for (record in allRows) {
-
         val id_String : String = record.values[0]
         val wannaDrive_String : String = record.values[1]
 
         val id_Int : Int = id_String.toInt()
         val wannaDrive_Boolean : Boolean = wannaDrive_String.toBoolean()
 
         driveInterest.add ( Vehicle(id = id_Int , wannaDrive = wannaDrive_Boolean ) )
-
     }
-
     return driveInterest
+}
+
+fun parseOutputToCSV (results : MutableList<Vehicle>) {
+    //not sure if this results in writing into the correct file by now
+    FileAccess().getWriter("/" + "results.csv")
+
+    val writer = CsvWriter(CsvWriterSettings())
+    // Write the record headers of this file
+    writer.writeHeaders("ID", "Wanted to Drive", "Delayed")
+
+    //problem with output: writer.write... seems not to offer any functions to print a (mutable)list to a csv;
+    //maybe transformation of list is necessary (to what? Array?)
+    val output = results
+
+    writer.writeRowsAndClose(output)
```

Comment:
Wie man eine Map draus macht hab ich jetzt nicht gefunden. Aber so gehts auch:

    val customerRows: MutableList<Array<Any>> = mutableListOf()
    for (result in results) {
        val id = result.id.toString()
        val delay = result.delayed.toString()
        val row: Array<Any> = arrayOf(id, delay)
        customerRows.add(row)
    }
    csvWriter.writeRowsAndClose(customerRows)

Zeile 87, 91, 93 würden damit ersetzt. 

Schade, konnte aus der Hilfe der Library ([hier](https://github.com/uniVocity/univocity-parsers#writing)) nicht schlau genug werden um es zu machen wie dort beschrieben. Konnte nur von woanders abschreiben.